RIOT-OS/RIOT
# rust: Make dependencies visible

### Contribution description

The Cargo.lock of rust_riotmodules_standalone should explicitly list all
    versions of software that would ever be pulled in by enabling RIOT
    modules backed by a Rust crate. This did not happen before, because
    while the optional dependencies could all be selected on the command
    line, they were not visible as possible features to `cargo update`, and
    were thus not encoded in the Cargo.lock file.

### Testing procedure

CI is probably good enough, but to understand what it does, look at `cargo tree --all-features` in `sys/rust_riotmodules_standalone` before and after -- and compare those with the output of `make` in `tests/rust_libs`. You will see that before, `cargo tree -all-features` did not show the `lsm303agr` crate as a listed dependency, even though the `make` output clearly illustrates that it *is* being pulled in. After, all crates used will also be shown in `cargo tree`.

These `cargo tree` outputs also align with the content of the Cargo.lock files before and after.

### Issues/PRs references

This should probably have been done right in #16833 -- sorry for the oversight.

This PR is built on #18683 -- so while that's not in, please ignore the first commit.
